bitwarden_license/test/Commercial.Core.Test/SecretsManager/AccessPolicies/DeleteAccessPolicyCommandTests.cs

```diff
@@ -11,28 +15,241 @@
+                mockServiceAccount.OrganizationId = grantedProject.OrganizationId;
+                policyToReturn = new ServiceAccountProjectAccessPolicy
+                {
+                    Id = data,
+                    GrantedProjectId = grantedProject.Id,
+                    ServiceAccount = mockServiceAccount,
+                    GrantedProject = grantedProject
+                };
+                break;
+        }
+
+        switch (permissionType)
+        {
+            case PermissionType.RunAsAdmin:
+                sutProvider.GetDependency<ICurrentContext>().OrganizationAdmin(grantedProject.OrganizationId)
+                    .Returns(true);
+                break;
+            case PermissionType.RunAsUserWithPermission:
+                sutProvider.GetDependency<IProjectRepository>().UserHasWriteAccessToProject(grantedProject.Id, userId)
+                    .Returns(true);
+                break;
+        }
+
+        sutProvider.GetDependency<IAccessPolicyRepository>().GetByIdAsync(data)
+            .Returns(policyToReturn);
+
+        await sutProvider.Sut.DeleteAsync(data, userId);
+
+        await sutProvider.GetDependency<IAccessPolicyRepository>().Received(1).DeleteAsync(Arg.Is(data));
+    }
+
+    [Theory]
+    [BitAutoData(AccessPolicyType.UserProjectAccessPolicy)]
+    [BitAutoData(AccessPolicyType.GroupProjectAccessPolicy)]
+    [BitAutoData(AccessPolicyType.ServiceAccountProjectAccessPolicy)]
+    public async Task DeleteAccessPolicy_UserProjectAccessPolicy_PermissionsCheck_ThrowsNotAuthorized(
+        AccessPolicyType accessPolicyType,
+        Guid data,
+        Guid userId,
+        Group mockGroup,
+        ServiceAccount mockServiceAccount,
+        Project grantedProject,
+        SutProvider<DeleteAccessPolicyCommand> sutProvider)
+    {
+        sutProvider.GetDependency<ICurrentContext>().AccessSecretsManager(Arg.Any<Guid>()).Returns(true);
+        BaseAccessPolicy policyToReturn = null;
+
+        switch (accessPolicyType)
+        {
+            case AccessPolicyType.UserProjectAccessPolicy:
+                policyToReturn = new UserProjectAccessPolicy { Id = data, GrantedProjectId = grantedProject.Id, GrantedProject = grantedProject };
+                break;
+            case AccessPolicyType.GroupProjectAccessPolicy:
+                policyToReturn =
+                    new GroupProjectAccessPolicy { Id = data, GrantedProjectId = grantedProject.Id, Group = mockGroup, GrantedProject = grantedProject };
+                break;
+            case AccessPolicyType.ServiceAccountProjectAccessPolicy:
+                policyToReturn = new ServiceAccountProjectAccessPolicy
+                {
+                    Id = data,
+                    GrantedProjectId = grantedProject.Id,
+                    ServiceAccount = mockServiceAccount,
+                    GrantedProject = grantedProject,
+                };
+                break;
+        }
```

Comment:
I think we can clean up the tests a bit by extracting this into a function? It seems to be mostly reused in `DeleteAccessPolicy_ProjectGrants_PermissionsCheck_Success`.

Reply:
Resolved in https://github.com/bitwarden/server/pull/2628/commits/79bce819b50d75b5ce021b4b4dfad21dc9c97cd8